packages/autoloader/src/AutoloadGenerator.php

```diff
@@ -313,42 +311,17 @@ private function getAutoloadFilesPackagesFile( $filesMap ) {
 	/**
 	 * Generate the PHP that will be used in the autoload_packages.php files.
 	 *
-	 * @param string $suffix  Unique suffix added to the jetpack_enqueue_packages function.
-	 *
 	 * @return string
 	 */
-	private function getAutoloadPackageFile( $suffix ) {
+	private function getAutoloadPackageFile() {
 		$sourceLoader   = fopen( __DIR__ . '/autoload.php', 'r' );
 		$file_contents  = stream_get_contents( $sourceLoader );
 		$file_contents .= <<<INCLUDE_FILES
 /**
  * Prepare all the classes for autoloading.
  */
-function enqueue_packages_$suffix() {
-	\$class_map = require_once dirname( __FILE__ ) . '/composer/autoload_classmap_package.php';
-	foreach ( \$class_map as \$class_name => \$class_info ) {
-		enqueue_package_class( \$class_name, \$class_info['version'], \$class_info['path'] );
-	}
-
-	\$autoload_file = __DIR__ . '/composer/autoload_files_package.php';
-
-	\$includeFiles = file_exists( \$autoload_file )
-		? require \$autoload_file
-		: array();
-
-	foreach ( \$includeFiles as \$fileIdentifier => \$file_data ) {
-		enqueue_package_file( \$fileIdentifier, \$file_data[ 'version' ], \$file_data[ 'path' ] );
-	}
-
-	if ( function_exists( 'has_action') && function_exists( 'did_action' ) && ! did_action( 'plugins_loaded' ) && false === has_action( 'plugins_loaded', __NAMESPACE__ . '\\file_loader' ) ) {
-		// Add action if it has not been added and has not happened yet.
-		// Priority -10 to load files as early as possible in case plugins try to use them during `plugins_loaded`.
-		add_action( 'plugins_loaded', __NAMESPACE__ . '\\file_loader', 0, -10 );
-	} elseif( ! function_exists( 'did_action' ) || did_action( 'plugins_loaded' ) ) {
-		file_loader(); // Either WordPress is not loaded or plugin is doing it wrong. Either way we'll load the files so nothing breaks.
-	}
-}
-enqueue_packages_$suffix();
+\$vendorPath = dirname( __FILE__ );
+require_once dirname( __FILE__ ) . '/automattic/jetpack-autoloader/src/function-enqueue-packages.php';
```

Comment:
I think that we'll need to set the path to `function-enqueue-packages.php` based on the location of the latest installed autoloader. Is that what you're thinking? It's not a blocker for merging this into the feature branch, I just want to make sure I understand the plan for this file :smile:

Reply:
That would be one way, yes, but I wanted to handle selecting the newest autoload inside the included file. Mostly because working inside a string is not very pleasant and is not subject to linter checks.

Reply:
I agree about the unpleasantness of working inside a string!

If we're going to select the newest autoloader in that file, we might want to put all of the code that’s currently in the file in a function. As it is, all of that code will be executed by the first autoloader that gets loaded. I think we need to isolate it so that only the latest autoloader does that work.